WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java

```diff
@@ -306,6 +307,36 @@ private void setHasFetchedMediaForAllFilters() {
         }
     }
 
+    /*
+        given a list of media, it rechecks its state against the UploadingService to see
+        if it's uploading or queued and change it accordingly, to recover from an inconsistent state
+     */
+    private List<MediaModel> getSanitizedMedia(List<MediaModel> uncuredList) {
```

Comment:
> since the upload states will remain incorrect until the user enters the media browser

I know, it was intended - we're actually delaying doing any processing until it is really needed, which is, when the user wants to see the media library. 

> Calling this routine every time the media grid is populated [...] seems unnecessary.

We're doing it every time such a list is needed, and we do that each time `getAdapter().setMediaList()` is called  as to avoid any "old copy" problem. We're modifying the objects and returning the list right away (the fact that we return the same List object is just an artifact to make it cleaner to make these chained calls).  With this approach, there is absolutely no way  the user is going to see something bad.

Maybe this looks like too much,  most of the times it won't be necessary and it will just leave without looping more than once, given there wouldn't be things to check for in the UploadService.

All that explained, if this still sounds/looks too crazy, I can try the approach of  
a) just sanitizing when the app starts, which will work for the scenario when the app crashed / is stopped and there's an inconsistent state in the MediaStore, but will leave any unforeseeable scenarios untackled, or 
b) I can also try sanitizing media when the Activity is created, pretty much the same thing we do when opening the Editor, which is "the other part of the app" when the user needs to see exactly what they have.

I know we're probably settling on b) but wanted to give you the full context of what I had in mind when sending this PR your way :)